demos/chat-client/c++/main.cpp

```diff
@@ -3,24 +3,129 @@
 #include <unistd.h>
 #include <string>
 #include <cstring>
+#include <sys/types.h>
+#include <sys/select.h>
+#include <netinet/in.h>
+#include <sys/socket.h>
+#include "assets.h"
+
+// C backend code based on https://github.com/rjzak/web_wordpuzzle
+
+const unsigned char HTTP_OK[14] = "HTTP/1.0 200\n";
+const unsigned char HTTP_NOT_FOUND[14] = "HTTP/1.0 404\n";
+const unsigned char CONTENT_TYPE_HTML[26] = "Content-Type: text/html\n\n";
+const unsigned char CONTENT_TYPE_PLAIN[27] = "Content-Type: text/plain\n\n";
+const unsigned char CONTENT_TYPE_JAVASCRIPT[32] = "Content-Type: text/javascript\n\n";
+
+const unsigned int PORT = 8080;
+
+ssize_t write_all(const int fd, const unsigned char *buf, const size_t n);
+
+// Pass --webinterface in Enarx.toml to use webinterface, uses std::in otherwise
+
+int main(int argc, char *argv[])
+{
+    if (argc > 2) {
+        std::cout << "Too many arguments provided" << std::endl;
+        return 1;
+    }
+
+    bool isWebInterface = false;
```

Comment:
I think this should just be enabled by default, we can add functionality to disable it later if we need it